sycl/test-e2e/Plugin/level_zero_queue_priority.cpp

```diff
@@ -1,12 +1,14 @@
 // REQUIRES: gpu, level_zero, level_zero_dev_kit
 // RUN: %{build} %level_zero_options -o %t.out
-// RUN: env ZE_DEBUG=1 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK-STD
-// RUN: env ZE_DEBUG=1 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK-IMM
+// RUN: env ZE_DEBUG=1 SYCL_ENABLE_DEFAULT_CONTEXTS=1 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK-STD
+// RUN: env ZE_DEBUG=1 SYCL_ENABLE_DEFAULT_CONTEXTS=1 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK-IMM
 //
 // Check that queue priority is passed to Level Zero runtime
 // This is the last value in the ZE_STRUCTURE_TYPE_COMMAND_QUEUE_DESC
 //
-// With immediate command lists the command lists are recycled between queues.
+// With immediate command lists the command lists are recycled between queues in
+// a context. We make sure all queues in this test use the same context by
+// explicitly specifying SYCL_ENABLE_DEFAULT_CONTEXTS=1

```

Comment:
Instead of relying on the default context, maybe it would make sense to pass the context and device to `test` and use them to create the queue? That said, maybe we want to make sure both cases work.

Reply:
The test was changed recently to check that immediate command lists are recycled within a context. How the queues share the same context is orthogonal to that. Using the env var to use the same context was simpler.

Reply:
There is really value in both ways of checking it. Personally I would prefer the explicit passing of context and device, just to make the test less tied to the default-context behavior, which is somewhat implicitly required for this test (except for the comment you add,) but either is fine with me.